src/main/java/bio/terra/common/db/JdbcConfiguration.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.common.db;
+
+import java.util.Properties;
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DriverManagerConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+public class JdbcConfiguration {
```

Comment:
Pulling this class seems will make the migration easier, and I think it makes sense to pull this class here.
For migrating client to using this base class, it does sound out of this ticket's scope. You can create few tickets to track clients migration for using this base class, and for this ticket, just use if for `StairwayJdbcConfiguration` migration.  `ServerJdbcConfiguration` migration should be straight forward, anyone can pick it up.

Reply:
One thing to note is that my comment re: JMX/PF-485 comes back into play if we go this route; if we consumed this as-is in Janitor it could re-introduce that defect there (for instance); so the follow-on ticket may then involve coming back to the common lib to add the configurable.

Reply:
Is the configuration we'd need something we can just add to the right properties file, or more interesting?

Reply:
The scope of the code change can be seen in this commit (add a property and use it to setup the `GenericObjectPoolConfig<>` passed to `GenericObjectPool <>` ctor):

https://github.com/DataBiosphere/terra-resource-janitor/pull/93/commits/331e57c8dcc7dd07471fab7d7a916f3b61ba64a4

Downstream consumers would be able to consume it as needed (note that the extending class is the one that sets the config, thus the changes to two different test configs in that commit).